src/platform/h563xx/adc_ll.c

```diff
@@ -45,16 +61,42 @@ void HAL_ADC_MspInit(ADC_HandleTypeDef *hadc)
     __HAL_RCC_ADC_CLK_ENABLE();
     // Enable GPIOA clock
     __HAL_RCC_GPIOA_CLK_ENABLE();
+    // Enable DMA1 clock
+    __HAL_RCC_GPDMA1_CLK_ENABLE();
 
     // Configure GPIO pin for ADC1_IN0 (PA0)
     gpio_init.Pin = GPIO_PIN_0; // PA0
     gpio_init.Mode = GPIO_MODE_ANALOG;
     gpio_init.Pull = GPIO_NOPULL;
     HAL_GPIO_Init(GPIOA, &gpio_init);
 
+    /*DMA for the ADC*/
+    g_hdma_adc.Instance = GPDMA1_Channel6; // DMA channel for ADC1
+    g_hdma_adc.Init.Request = GPDMA1_REQUEST_ADC1;
+    g_hdma_adc.Init.BlkHWRequest = DMA_BREQ_SINGLE_BURST;
+    g_hdma_adc.Init.Direction = DMA_PERIPH_TO_MEMORY;
+    g_hdma_adc.Init.SrcInc = DMA_SINC_FIXED;
+    g_hdma_adc.Init.DestInc = DMA_DINC_INCREMENTED;
+    g_hdma_adc.Init.SrcDataWidth = DMA_SRC_DATAWIDTH_HALFWORD;
+    g_hdma_adc.Init.DestDataWidth = DMA_SRC_DATAWIDTH_HALFWORD;
```

Comment:
```suggestion
    g_hdma_adc.Init.DestDataWidth = DMA_DEST_DATAWIDTH_HALFWORD;
```

This makes the output sane:

[INFO ] Sample 1: 3040
[INFO ] Sample 2: 3040
[INFO ] Sample 3: 3040
[INFO ] Sample 4: 3040